### ISSUE TITLE
Deprecate jackknife in favour of astropy's?

### ISSUE THREAD

**Issue by bsipocz** [astroML/astroML#156]
This needs some investigation and also benchmarking whether it makes sense to deprecate `astroML.resample.jackknife` in favour of `astropy.stats.jackknife_resampling` and `astropy.stats.jackknife_stats` or there is an argument to keep it at both places.

cc @mirca in case he wants to comment from the astropy side

**Comment by mirca** [comment]
Hi @bsipocz! I will try to be as unbiased as possible on this.

My take on this is that the ``astropy`` implementation, while being a little bit faster than ``astroML``'s, also has a simpler API with additional features, e.g., confidence interval at an arbitrary level.  See the code snippet below.

```
In [1]: from astroML.resample import jackknife

In [2]: from astropy.stats import jackknife_stats

In [3]: import numpy as np

In [4]: data = np.random.normal(size=int(1e4))

In [5]: %timeit jackknife_stats(data, np.mean)
1.87 s ± 624 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [6]: %timeit jackknife(data, np.mean, kwargs=dict(axis=1))
5.77 s ± 989 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [7]: jackknife_stats(data, np.mean)
Out[7]:
(0.01485987260643617,
 0.0,
 0.0099542101368651111,
 array([-0.00465002,  0.03436977]))

In [8]: jackknife(data, np.mean, kwargs=dict(axis=1))
Out[8]: (0.01485987260643617, 0.0099532147656175001)
```

**Comment by ivezic** [comment]
@bsipocz We adopted a general principle in such cases that we should switch to astropy
implementation, unless it is technically inferior (functionality, speed, memory 
usage, in this priority order) to astroML implementation. Given the analysis above kindly 
provided by @mirca it seems that we should adopt astropy implementation here.

**Comment by bsipocz** [comment]
Fair enough. The advantage of astroML's version is the very convenient way of passing kwargs to the involved function without hacking with a lambda functions.

But that's pretty easy to port over, unless there is a good reason against it.

**Comment by bsipocz** [comment]
Ohh, and the option of returning the raw distribution. We use it in one figure, I'll look around on github whether anyone else uses it for anything.

**Comment by mirca** [comment]
@bsipocz should we propose those changes to the astropy implementation?

**Comment by bsipocz** [comment]
OK, other than various copies and renames of the figures, and meetups and reading groups working through the book I didn't find a use case for it in production code.

**Comment by bsipocz** [comment]
Yeah, I suppose we should. It a sensible feature, so I don't expect any pushbacks.
